src/libproc_macro/lib.rs

```diff
@@ -685,7 +685,7 @@ impl TokenTree {
                 })
             }
 
-            DotEq => unreachable!(),
+            DotEq => joint!('.', Eq),
```

Comment:
A tiny nit, but I'd move this line up to below `DotDotEq` so it's in a reasonable location.